VPLAY-10085 - AV freeze in live-pause-seekToLive scenario

anjali-syna opened the pull request:
AV freeze is observed in L2 tests while performing  seek to live after live-pause. Fixing this.

Changes:
* Setting 'mLocalAAMPInjectionEnabled' flag to true when TuneHelper is
* called with tuneType 'eTUNETYPE_SEEKTOLIVE' and there is data
* available in the TSB. This enables AampTsbReader to read and feed
* the available data in TSB to gstreamer

Test Procedure: Run L2 tests 5001 and 5002

anjali-syna committed:
VPLAY-10085 [ENT-4082] AV freeze in live-pause-seekToLive scenario

Reason for change: AV freeze is observed in L2 tests while performing
  seek to live after live-pause.

Changes:
* Setting 'mLocalAAMPInjectionEnabled' flag to true when TuneHelper is
* called with tuneType 'eTUNETYPE_SEEKTOLIVE' and there is data
* available in the TSB. This enables AampTsbReader to read and feed
* the available data in TSB to gstreamer

Test Procedure: Run L2 tests 5001 and 5002

anjali-syna committed:
VPLAY-10085 [ENT-4082] AV freeze in live-pause-seek to live scenario

Reason for change: AV freeze is observed in L2 tests while performing
  seek to live after live-pause.

Changes:
* Setting 'mLocalAAMPInjectionEnabled' flag to true when TuneHelper is
* called with tuneType 'eTUNETYPE_SEEKTOLIVE' and there is data
* available in the TSB. This enables AampTsbReader to read and feed
* the available data in TSB to gstreamer

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10085

jfagunde reviewed (commented):
These changes look good to me.

dnfaulkner commented on priv_aamp.cpp:
```diff
@@ -5276,10 +5276,21 @@ void PrivateInstanceAAMP::TuneHelper(TuneType tuneType, bool seekWhilePaused)
+		{
+			SetLocalAAMPTsbInjection(true);
+		}
+		else if( (NULL != tsbSessionManager) && (tsbSessionManager->GetTotalStoreDuration(eMEDIATYPE_VIDEO) > 0 ))
```
Shouldn't this if statement just be in an additional OR condition ('||') added to line 5286, as the outcome is to do the same thing?

dnfaulkner commented on priv_aamp.cpp:
```diff
@@ -5276,10 +5276,21 @@ void PrivateInstanceAAMP::TuneHelper(TuneType tuneType, bool seekWhilePaused)
+
+		if(tuneType != eTUNETYPE_SEEKTOLIVE)
+		{
+			SetLocalAAMPTsbInjection(true);
```
Does doing this make this line redundant or does it still serve the required purpose:
https://github.com/rdkcentral/aamp/blob/develop/priv_aamp.cpp#L11820
?

dnfaulkner commented on priv_aamp.cpp:
```diff
@@ -5276,10 +5276,21 @@ void PrivateInstanceAAMP::TuneHelper(TuneType tuneType, bool seekWhilePaused)
-	// any reason other than a new tune or seek to live (set rate, seek...)
-	if (!newTune && IsLocalAAMPTsb() && (tuneType != eTUNETYPE_SEEKTOLIVE))
+	// any reason other than a new tune or seek to live (set rate, seek...).
+	// Also, set LocalAAMPTsbInjection to true if tuneType is SEEKTOLIVE, and Tsb is not empty
```
Do you mean "...or AAMP TSB is not empty?

dnfaulkner reviewed (commented):
.

anjali-syna committed:
VPLAY-10085 [ENT-4082] AV freeze in live-pause-seek to live scenario

Reason for change: AV freeze is observed in L2 tests while performing
  seek to live after live-pause.

Changes:
* Setting 'mLocalAAMPInjectionEnabled' flag to true when TuneHelper is
* called with tuneType 'eTUNETYPE_SEEKTOLIVE' and there is data
* available in the TSB. This enables AampTsbReader to read and feed
* the available data in TSB to gstreamer

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
VPLAY-10085 [ENT-4082] AV freeze in live-pause-seek to live scenario

Reason for change: AV freeze is observed in L2 tests while performing
  seek to live after live-pause.

Changes:
* Setting 'mLocalAAMPInjectionEnabled' flag to true when TuneHelper is
* called with tuneType 'eTUNETYPE_SEEKTOLIVE' and there is data
* available in the TSB. This enables AampTsbReader to read and feed
* the available data in TSB to gstreamer

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Merge branch 'feature/VPLAY-10085' of github.com:rdkcentral/aamp into feature/VPLAY-10085

anjali-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10085

dnfaulkner commented on priv_aamp.cpp:
```diff
@@ -5276,10 +5276,17 @@ void PrivateInstanceAAMP::TuneHelper(TuneType tuneType, bool seekWhilePaused)
-		SetLocalAAMPTsbInjection(true);
+		AampTSBSessionManager *tsbSessionManager = GetTSBSessionManager();
+
+		if( (tuneType != eTUNETYPE_SEEKTOLIVE) || ((tuneType == eTUNETYPE_SEEKTOLIVE) && (NULL != tsbSessionManager) && (tsbSessionManager->GetTotalStoreDuration(eMEDIATYPE_VIDEO) > 0 ) ) )
```
You don't need to check (tuneType == eTUNETYPE_SEEKTOLIVE) because it's already determined that (tuneType != eTUNETYPE_SEEKTOLIVE) is false which means that tuneType == eTUNETYPE_SEEKTOLIVE

anjali-syna replied:
Done

dnfaulkner reviewed (commented):
Just one more comment

anjali-syna committed:
VPLAY-10085 [ENT-4082] AV freeze in live-pause-seek to live scenario

Reason for change: AV freeze is observed in L2 tests while performing
  seek to live after live-pause.

Changes:
* Setting 'mLocalAAMPInjectionEnabled' flag to true when TuneHelper is
* called with tuneType 'eTUNETYPE_SEEKTOLIVE' and there is data
* available in the TSB. This enables AampTsbReader to read and feed
* the available data in TSB to gstreamer

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Signed-off-by: anjali-syna <206662904+anjali-syna@users.noreply.github.com>

anjali-syna committed:
VPLAY-10085 [ENT-4082] AV freeze in live-pause-seek to live scenario

Reason for change: AV freeze is observed in L2 tests while performing
  seek to live after live-pause.

Changes:
* Setting 'mLocalAAMPInjectionEnabled' flag to true when TuneHelper is
* called with tuneType 'eTUNETYPE_SEEKTOLIVE' and there is data
* available in the TSB. This enables AampTsbReader to read and feed
* the available data in TSB to gstreamer

Test Procedure: Run L2 tests 5001 and 5002

Risks: Low

Merge branch 'feature/VPLAY-10085' of github.com:rdkcentral/aamp into feature/VPLAY-10085

anjali-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10085

anjali-syna committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10085

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-10085
